### ISSUE TITLE
Plack::Builder DSL and mount calls

### ISSUE THREAD

**Issue by stevan** [plack/Plack#531]
We would like to do the following:
```
builder {
    enable 'SomeMiddleware' => (blah => 1);
    foreach my $url ( keys %url_map ) {
        mount $url => $url_map{ $url };
    }
};
```
But Plack::Builder does not like this because there is no return value.

We would like to propose one edition to the logic inside `builder`:

> if the `builder` returns nothing (either `undef` or `()`) and mount has been called, just use `$urlmap`

The other logic and error checking inside `builder` should still work fine, and this should (mostly) be backwards compatible. 

The code changes within `Plack::Builder::builder` should basically be this:
```
my $app = $block->();

if ($mount_is_called) {
    $app ||= $urlmap; # <<< just adding this line in ... nothing more
    if ($app ne $urlmap) {
        Carp::carp("WARNING: You used mount() in a builder block, but the last line (app) isn't using mount().\n" .
                   "WARNING: This causes all mount() mappings to be ignored.\n");
    } else {
        $app = $app->to_app;
    }
}
```
If this is okay with you, we can prepare a formal pull request.

**Comment by miyagawa** [comment]
Can't you either have a fallback app outside the `for` loop, or do:

```perl
my $rv;
for (...) {
    $rv = mount ...;
}
$rv;
```

I understand that's arguably ugly, but just wanted to make sure that we're on the same page.

**Comment by stevan** [comment]
Yes, that would work, but (as you said) is ugly.

To be clear, this is not an giant issue or anything, we were just discussing this at work and realized that this simple case (we believe) is unambiguous, so why not handle it.

**Comment by miyagawa** [comment]
Yeah, i think defaulting to `$urlmap` when the return value is empty is sensible.

**Comment by stevan** [comment]
:)

**Comment by stevan** [comment]
Would you like a PR with tests and such? I would be happy to provide it.

**Comment by miyagawa** [comment]
PR with tests and documentation = :100:

**Comment by stevan** [comment]
*sigh* ... okay, docs too.

Look for something either later tonight or first thing tomorrow (EU) morning.
